File: ios/Classes/ShipbookFlutterOmnysPlugin.swift
import Flutter
import ShipBookSDK
import UIKit

public class ShipbookFlutterOmnysPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(name: "shipbook_flutter_omnys", binaryMessenger: registrar.messenger())
        let instance = ShipbookFlutterOmnysPlugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
    }
    
    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        let arguments = call.arguments as! NSArray
        switch call.method {
        case "Shipbook.start":
            let appId = arguments[0] as! String
            let appKey = arguments[1] as! String
            ShipBook.start(appId: appId, appKey: appKey)
            result(true)
        case "Shipbook.log":
            let tag = arguments[0] as! String
            let message = arguments[1] as! String
            //TODO receive severity from Dart code
            ShipBook.getLogger(tag).message(msg: message, severity: Severity.Debug)
            result(true)
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}

File: android/src/main/kotlin/com/omnys/flutter/shipbook/shipbook_flutter_omnys/ShipbookFlutterOmnysPlugin.kt
package com.omnys.flutter.shipbook.shipbook_flutter_omnys

import android.app.Application
import androidx.annotation.NonNull

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result
import io.shipbook.shipbooksdk.Models.Severity
import io.shipbook.shipbooksdk.ShipBook

/** ShipbookFlutterOmnysPlugin */
class ShipbookFlutterOmnysPlugin : FlutterPlugin, MethodCallHandler {
    /// The MethodChannel that will the communication between Flutter and native Android
    ///
    /// This local reference serves to register the plugin with the Flutter Engine and unregister it
    /// when the Flutter Engine is detached from the Activity
    private lateinit var channel: MethodChannel

    override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "shipbook_flutter_omnys")
        channel.setMethodCallHandler(this)
        flutterPluginBinding.applicationContext
    }

    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
        val args = call.arguments<ArrayList<*>>()

        when (call.method) {
            "Shipbook.start" -> {
                val appId = args!![0] as String
                val appKey = args[1] as String
                ShipBook.start(application, appId, appKey)
                result.success(true)
            }

            "Shipbook.log" -> {
                val tag = args!![0] as String
                val message = args[1] as String
                //TODO receive severity from Dart code
                ShipBook.getLogger(tag).message(message, Severity.Debug)
            }

            else -> result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }

    companion object {
        private lateinit var application: Application

        fun init(application: Application) {
            this.application = application
        }
    }
}
